Salwador/ViewModels/ImageGenerationViewModel.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  IGViewModel.swift
+//  Salwador
+//
+//  Created by Ailiniyazi Maimaiti on 16.02.23.
+//
+
+import Foundation
+import UIKit
+
+extension ImageGenerationView {
+    @MainActor class ImageGenerationViewModel: ObservableObject {
+        @Published var buttonText: String = "Ready?"
+        @Published var imagePath: String = "salvador-man"
+        @Published var image: UIImage? = nil
+        @Published var isLoading: Bool = false
+        @Published var hasError: Bool = false
+        @Published var errorMsg: String = ""
+
+        func sendRequest(prompText: String) async {
+            let fileManager = FileManagerService(filename: "APIKey", filetype: "json")
+
+            let data = fileManager.readFile()
+
+            do {
+                let json = try JSONDecoder().decode(APIJSONModel.self, from: data!)
+                let response = try await ImageGenerationService.shared.generateImage(withPrompt: prompText, apiKey: json.apikey)
+
+                if let url = response.data.map(\.url).first {
+                    let (data, _) = try await URLSession.shared.data(from: url)
+
+                    if data.isEmpty {
+                        hasError = true
+                        errorMsg = "No data was fetched, please retry again!"
+                    }
+
+                    image = UIImage(data: data)
+                    isLoading = false
+                    buttonText = "Regenerate"
+                }
+            } catch {
+                hasError = true
+                errorMsg = error.localizedDescription
+                isLoading = false
+                buttonText = "Regenerate"
+                print(error)
```

Comment:
suggestion: All of this can be part of a very expressive enum state as mentioned above.